waiter/src/waiter/state.clj

```diff
@@ -80,43 +111,59 @@
         expired-instances? (some (fn [[_ {:keys [status-tags]}]] (contains? status-tags :expired)) instance-id->state)
         starting-instances? (some (fn [[_ {:keys [status-tags]}]] (contains? status-tags :starting)) instance-id->state)]
     (->> instance-id->state
-         (filter (fn [[_ state]] (killable? state)))
+         (filter (fn [[instance-id state]]
+                   (-> (instance-id->request-id->use-reason-map instance-id)
+                       (killable? earliest-request-threshold-time state))))
          (filter (fn [[instance-id _]] (acceptable-instance-id? instance-id)))
+         seq ;; trigger short-circuit if there are no expired instances
```

Comment:
Is this comment right? What is this `seq` doing?

Reply:
Ugh, changing it to `some->>` to ensure the comment stays right